### PR TITLE
retry on ClientErrors in request_cdn_cache_invalidation

### LINKED ISSUE #6177: ClientError (Throttling) in request_cdn_cache_invalidation()
https://sentry.prod.mozaws.net/operations/mdn-prod/issues/6766908/

```
ClientError: An error occurred (Throttling) when calling the CreateInvalidation operation (reached max retries: 4): Rate exceeded
(1 additional frame(s) were not displayed)
...
  File "newrelic/hooks/application_celery.py", line 85, in wrapper
    return wrapped(*args, **kwargs)
  File "celery/app/trace.py", line 648, in __protected_call__
    return self.run(*args, **kwargs)
  File "kuma/api/tasks.py", line 180, in request_cdn_cache_invalidation
    invalidation = client.create_invalidation(
  File "botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "botocore/client.py", line 661, in _make_api_call
    raise error_class(parsed_response, operation_name)

ClientError: An error occurred (Throttling) when calling the CreateInvalidation operation (reached max retries: 4): Rate exceeded
```

### PR DESCRIPTION
**Pull request by peterbe**
Fixes #6177

I started a `shell_plus` and ran:
```
>>> [request_cdn_cache_invalidation.delay([(x.locale, x.slug) for x in Document.objects.filter(is_redirect=False).only('locale', 'slug').order_by('?')[:100]]) for _ in range(31)]
```
Eventually, it got the same error as mentioned in Sentry. 
Then I applied the fix. (I had a print statement too inside the function) and ran it again. 
Now I can see that it slept and retried but at least it didn't error.

### CODE REVIEW

**Comment by escattone** [comment]
This looks great. I'm just waiting for @peterbe to resolve the merge conflict by using our new `poetry` package manager instead.

File: pyproject.toml [hunk 1]
```diff
@@ -105,6 +105,7 @@ py = "1.4.33"
 python3-openid = "3.1.0"
 requests-oauthlib = "0.8.0"
 webencodings = "0.5.1"
+redo = "^2.0.3"
```
Comment (peterbe) [inline pyproject.toml]: Did I do it right @callahad ?

I used `docker-compose exec web poetry add redo`

Comment (callahad) [reply pyproject.toml]: Yep, that looks totally reasonable on the Poetry side

Comment (peterbe) [reply pyproject.toml]: Thanks. I'll wait for @escattone to review and merge.

**Pull request merged**